Sources/OuterCore/Scope.swift

```diff
@@ -73,4 +73,26 @@ final class Scope {
       }
     }
   }
+
+  public static func == (lhs: Scope, rhs: Scope) -> Bool {
+    return ObjectIdentifier(lhs) == ObjectIdentifier(rhs)
```

Comment:
You can delegate the equality test to the entry continuation instead of going through ObjectIdentifier.  That should always be unique (an invariant for the verifier as well, I suppose).